java/statestore/src/main/java/sleeper/statestore/dynamodb/DynamoDBStateStore.java

```diff
@@ -37,15 +37,21 @@ public class DynamoDBStateStore extends DelegatingStateStore {
     public static final String PARTITION_ID = DynamoDBPartitionFormat.ID;
     public static final String PARTITION_ID_AND_FILENAME = DynamoDBFileInfoFormat.PARTITION_ID_AND_FILENAME;
     public static final String TABLE_ID = "TableId";
+    public static final int DEFAULT_PAGE_LIMIT = 1024 * 1024;
```

Comment:
This default is a bit misleading since it looks like it's supposed to be equivalent to the DynamoDB default max page size of 1MB. This value is actually used to limit the number of items in a page rather than the number of bytes.

The built-in 1MB limit is a hard limit on the raw amount of data in a page of query results, regardless of the limit we set on the query.

If we default the limit to null, it would have the same effect.